<?xml version="1.0"?>
<robot name="ranger_mini" xmlns:xacro="http://www.ros.org/wiki/xacro">

<!-- Gazebo references -->

<gazebo reference="base_link">
  <mu1 value="0.1"/>
  <mu2 value="0.1"/>
  <!-- <kp  value="10000000.0" />
  <kd  value="1.0" /> -->
  <material>Gazebo/White</material>
</gazebo>

	<gazebo reference="front_left_steering_link">
	  <!-- <selfCollide>false</selfCollide> -->
      <mu1 value="0.5" />
      <mu2 value="0.5" />
     <material>Gazebo/Grey</material>
    </gazebo>

	<gazebo reference="front_right_steering_link">
	  <!-- <selfCollide>false</selfCollide> -->
      <mu1 value="0.5" />
      <mu2 value="0.5" />
      <material>Gazebo/Grey</material>
    </gazebo>
  
  	<gazebo reference="rear_left_steering_link">
	  <!-- <selfCollide>false</selfCollide> -->
      <mu1 value="1.5" />
      <mu2 value="1.5" />
      <material>Gazebo/Grey</material>
    </gazebo>

  	<gazebo reference="rear_right_steering_link">
	  <!-- <selfCollide>false</selfCollide> -->
      <mu1 value="1.5" />
      <mu2 value="1.5" />
      <material>Gazebo/Grey</material>
    </gazebo>

    <gazebo reference="ffl_wheel_link">
      <!-- <selfCollide>false</selfCollide> -->
    <mu1>0.8</mu1>
    <mu2>0.8</mu2>
    <kp>10000000.0</kp>
    <kd>1.0</kd>
    <minDepth>0.001</minDepth>
    <maxVel>0.1</maxVel>
    <fdir1>1 0 0</fdir1>
      
      <material>Gazebo/Grey</material>
    </gazebo>

    <gazebo reference="ffr_wheel_link">
      <!-- <selfCollide>false</selfCollide> -->
      <mu1>0.8</mu1>
    <mu2>0.8</mu2>
    <kp>10000000.0</kp>
    <kd>1.0</kd>
    <minDepth>0.001</minDepth>
    <maxVel>0.1</maxVel>
    <fdir1>1 0 0</fdir1>
      <material>Gazebo/Grey</material>
    </gazebo>

    <gazebo reference="rfl_wheel_link">
      <!-- <selfCollide>false</selfCollide> -->
      <mu1>0.8</mu1>
    <mu2>0.8</mu2>
    <kp>10000000.0</kp>
    <kd>1.0</kd>
    <minDepth>0.001</minDepth>
    <maxVel>0.1</maxVel>
    <fdir1>1 0 0</fdir1>
      <material>Gazebo/Grey</material>
    </gazebo>

    <gazebo reference="rfr_wheel_link">
      <!-- <selfCollide>false</selfCollide> -->
      <mu1>0.8</mu1>
    <mu2>0.8</mu2>
    <kp>10000000.0</kp>
    <kd>1.0</kd>
    <minDepth>0.001</minDepth>
    <maxVel>0.1</maxVel>
    <fdir1>1 0 0</fdir1>
      <material>Gazebo/Grey</material>
    </gazebo>

    <!-- ........................... CAMERA PLUGIN ................................... -->

    <gazebo reference="kinect_camera">
        <sensor name="kinect_camera" type="depth">
            <pose> 0 0 0 0 0 0 </pose>
            <visualize>true</visualize>
            <update_rate>30</update_rate>
            <camera>
                <horizontal_fov>1.089</horizontal_fov>
                <image>
                    <format>R8G8B8</format>
                    <width>640</width>
                    <height>480</height>
                </image>
                <clip>
                    <near>0.05</near>
                    <far>8.0</far>
                </clip>
            </camera>
            <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
                <ros>
                    <namespace>/</namespace>
                </ros>
                <frame_name>kinect_camera_optical</frame_name>
                <min_depth>0.1</min_depth>
                <max_depth>100</max_depth>
            </plugin>
        </sensor>
    </gazebo>
    

</robot>
 
